streams/src/main/java/org/apache/kafka/streams/processor/internals/Tasks.java

```diff
@@ -316,6 +315,14 @@ StreamsProducer threadProducer() {
         return activeTaskCreator.producerClientIds();
     }
 
+    Consumer<byte[], byte[]> mainConsumer() {
```

Comment:
Not a modification suggestion: for main consumer though, compared to topologymetadata below I think it's better indeed to expose it instead of passing in to `TaskExecutor` since we may have the consumer to be owned by a different thread in the future and committing would not be executed by the stream thread. So keeping here is preferred.